### DISCUSSION TITLE
CURL as a command line tool WebSocket

### DISCUSSION THREAD

**Discussion by nbanb** [curl/curl#13256] [General]
Hi 

it would be very helpfull to use 'curl' as a websocket command line tool like tools like Websocat is (https://github.com/vi/websocat)

From the actual documentation, I have seen since more than 1 year that "That part of the work has not been started".

Does someone know when it would be planned ?
What to do to make that part of the job starting in a reasonable time ?

I'm really interrested in the idea curl can act as websocat (need it for my bash library) 

Don't know if there is something I can do to make it goes forward, but let me know if I can help (testing? I'm not dev but maybe I could try coding ?).

Kind regards
nbanba

**Comment by bagder** [comment]
We welcome all help and participation that would bring that forward. We could start with ideas and descriptions on how it should work and then continue on to make sure it also works like that.

Personally I have not put a lot of focus on the WebSocket code recently mostly because of lack of time. But this is why we are open source: everyone can join in and help and pull the parts forward that need pulling.
